### ISSUE TITLE
macOS 10.15.5: tab is autoclosed after launch command executes in startup session script

### ISSUE THREAD

**Issue by TheJavaGuy** [kovidgoyal/kitty#2861]
**Describe the bug**
TL;DR Tab disappears if anything is written after `launch` in startup session file.

In more details: I start kitty with a startup session using command `kitty --session ~/.config/kitty/bug.session`. The content of `bug.session` file is:

```
# tab 1 - main repo
layout tall
# Set the working directory for windows in the current tab
cd ~
# Create a window and run the specified command in it
title tab1
launch

# tab 2 - home
new_tab home
cd ~
title tab2
launch brew update

```

On kitty startup, both tab1 and tab2 are shown. But after about 2 seconds tab2 disappears! It executes `brew update` and is immediately closed. Same thing happens if instead of `brew update` I write `git fetch` or some other command.

**Expected behavior**
Tab shouldn't disappear after executing command.

**Environment details**
OS: MacOS Catalina 10.15.5 on MacBook Pro

Output of kitty --debug-config:
```
kitty 0.18.1 created by Kovid Goyal
Darwin CFMAC-0027.local 19.5.0 Darwin Kernel Version 19.5.0: Tue May 26 20:41:44 PDT 2020; root:xnu-6153.121.2~2/RELEASE_X86_64 x86_64
ProductName:	Mac OS X ProductVersion:	10.15.5 BuildVersion:	19F101
Loaded config files: /Users/ivanmilosavljevic/.config/kitty/kitty.conf

Config options different from defaults:
active_tab_font_style (True, False)
active_tab_foreground Color(red=0, green=102, blue=0)
allow_remote_control  y
copy_on_select        clipboard
cursor_blink_interval 0.0
draw_minimal_borders  False
font_size             13.0
inactive_text_alpha   0.7
initial_window_height (60, 'cells')
initial_window_width  (160, 'cells')
input_delay           4
remember_window_size  False
tab_bar_style         separator
tab_fade              (0.5, 1.0)
tab_title_template    {index}: {title}
Added shortcuts:
	shift+backspace KeyAction(func='send_text', args=['application', b'\x1b[1;4R'])
	control+page_up KeyAction(func='previous_tab', args=())
	control+page_down KeyAction(func='next_tab', args=())
	super+backspace KeyAction(func='send_text', args=['application', b'\x15'])
	super+right KeyAction(func='send_text', args=['application', b'\x1bOF'])
	super+left KeyAction(func='send_text', args=['application', b'\x1bOH'])
	shift+super+backspace KeyAction(func='send_text', args=['application', b'\x0b'])
Changed shortcuts:
	super+1 KeyAction(func='goto_tab', args=(1,))
	super+2 KeyAction(func='goto_tab', args=(2,))
	super+3 KeyAction(func='goto_tab', args=(3,))
	super+4 KeyAction(func='goto_tab', args=(4,))
	super+5 KeyAction(func='goto_tab', args=(5,))
	super+6 KeyAction(func='goto_tab', args=(6,))
	super+7 KeyAction(func='goto_tab', args=(7,))
	super+8 KeyAction(func='goto_tab', args=(8,))
	super+9 KeyAction(func='goto_tab', args=(9,))
```

**Additional context**
Behavior is the same even if started without config, i.e. `kitty --session ~/.config/kitty/bug.session --config NONE`

**Comment by Luflosi** [comment]
See #2559.

**Comment by TheJavaGuy** [comment]
@Luflosi I saw that but unfortunately it's of little help.

If I try `launch zsh -o interactive "brew update; zsh"` it still autocloses the tab. 

If I open another tab before the current one then I keep the same directory (which is OK), but I can't see the result of command (because the tab autocloses, which isn't OK).

**Comment by kovidgoyal** [comment]
launch sh -c "brew update; read"
